tests/frameworks/test_django.py

```diff
@@ -4,109 +4,114 @@
 import os
 
 import urllib3
+import pytest
+from typing import Generator
 from django.apps import apps
 from django.contrib.staticfiles.testing import StaticLiveServerTestCase
 
-from ..apps.app_django import INSTALLED_APPS
+from tests.apps.app_django import INSTALLED_APPS
 from instana.singletons import agent, tracer
-from ..helpers import fail_with_message_and_span_dump, get_first_span_by_filter, drop_log_spans_from_list
+from tests.helpers import fail_with_message_and_span_dump, get_first_span_by_filter, drop_log_spans_from_list
+from instana.instrumentation.django.middleware import url_pattern_route
 
 apps.populate(INSTALLED_APPS)
 
 
 class TestDjango(StaticLiveServerTestCase):
-    def setUp(self):
+    @pytest.fixture(autouse=True)
+    def _resource(self) -> Generator[None, None, None]:
         """ Clear all spans before a test run """
-        self.recorder = tracer.recorder
-        self.recorder.clear_spans()
         self.http = urllib3.PoolManager()
+        self.recorder = tracer.span_processor
+        self.recorder.clear_spans()
 
-    def tearDown(self):
+    def tearDown(self) -> None:
```

Comment:
Is this method being used anywhere? If not, you can remove it or move it to the `_resource()` one.

Reply:
Moved it to `_resource()`!